pkg/arvo/sys/vane/ames.hoon

```diff
@@ -1729,14 +1739,33 @@
     ::
     =.  todos             (mutate todos)
     =.  peers.ames-state  (~(put by peers.ames-state) ship %alien todos)
-    ::  ask jael for .sndr life and keys on first contact
-    ::
     ?:  already-pending
       event-core
-    ::  NB: we specifically look for this wire in +public-keys-give in
-    ::  Jael.  if you change it here, you must change it there.
+    ::  ask Jael for (non-comet) ship's life and keys
     ::
-    (emit duct %pass /public-keys %j %public-keys [n=ship ~ ~])
+    ?.  =(%pawn (clan:title ship))
+      ::  NB: we specifically look for this wire in +public-keys-give in
+      ::  Jael.  if you change it here, you must change it there.
+      ::
+      (emit duct %pass /public-keys %j %public-keys [n=ship ~ ~])
+    (request-attestation ship)
+  ::  +request-attestation: helper to request attestation from comet
+  ::
+  ::    Comets will respond to any unknown peer with a self-attestation,
+  ::    so we either send a sendkeys packet (a dummy shut packet) or, if
+  ::    we're a comet, our own self-attestation, saving a roundtrip.
+  ::
+  ::    Also sets a timer to resend the request every 30s.
+  ::
+  ++  request-attestation
+    |=  [=ship]
+    ^+  event-core
+    =/  packet  ?.  =(%pawn (clan:title our))
+                  (sendkeys-packet ship)
+                (attestation-packet ship 1)
+    =.  event-core  (send-blob | ship packet)
```

Comment:
is there a cleaner way to write this?

Reply:
I don't think so. Both "actions" (`+send-blob` and `+emit`) produce the whole event-core, and dispatching each requires local state, so you can't easily sequence them with `=>`. You could build some kind of helper core hear, but this seems fine.